constraint-solver/src/solver/exhaustive_search.rs

```diff
@@ -69,7 +69,15 @@ pub fn get_brute_force_candidates<'a, T: RuntimeConstant, V: Clone + Hash + Ord>
     rc: impl RangeConstraintProvider<T::FieldType, V> + Clone + 'a,
 ) -> impl Iterator<Item = BTreeSet<V>> + 'a {
     constraint_system
-        .expressions()
```

Comment:
This is the only place where we use `expressions()` for a different purpose than figuring out the variables and here, it is a bit tricky, since we rely on `expressions()` to return the outermost layer of the expression structure only. I think it is better to do that explicitly by accessing `c.expression` and `b.fields()`.